src/invoice.rs

```diff
@@ -135,19 +138,55 @@ pub enum InvoiceParseError {
     IfaceName(InvalidIdent),
 }
 
+fn percent_decode(estr: &EStr) -> Result<String, InvoiceParseError> {
+    Ok(estr
+        .decode()
+        .into_string()
+        .map_err(|e| InvoiceParseError::InvalidQueryParam(e.to_string()))?
+        .to_string())
+}
+
+impl RgbInvoice {
+    fn map_query_params(uri: &Uri<&str>) -> Result<IndexMap<String, String>, InvoiceParseError> {
```

Comment:
nit: we do not use `self` here so no point of having this function inside of `RgbInvoice`. I think we can move it as `percent_decode` to be standalong